### PR TITLE
[risk=no][RW-11144] Fix workspace links in Cloud Environments Page

### PR DESCRIPTION
**Pull request by dmohs**
It turns out that the Cloud Environments Page Terra UI component uses labels to associate runtimes returned by Leonardo with workspaces. Without this association, the behavior of the component is degraded. This change adds the same labels when we create a runtime so this association can happen.

A similar solution should work to improve the behavior of the display of persistent disks, but that is not addressed by this change.

Tested manually by running a local UI against a local API and observing the difference. Clicking on a link to a workspace does open the correct workspace.

_Note: Any runtimes created before this PR is merged will continue to be displayed in a degraded format._

Before:
before

After:
linked

---
**PR checklist**

- [X] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [X] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [X] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [X] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [X] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

**Review by Qi77Qi — APPROVED** [review]
haven't tested out yet myself...but the change looks good to me 👍 (and seems pretty low risk to merge)

File: ui/src/app/pages/runtimes-list.tsx [hunk 1]
```diff
@@ -132,7 +137,8 @@ export const RuntimesList = fp.flow(
             <Environments
               {...{
                 nav: {
-                  getLink: (_, { namespace }) => `/workspaces/${namespace}`,
+                  getLink: (_, { namespace, name }) =>
+                    `/workspaces/${namespace}/${stringToSlug(name)}/data`,
```
Comment (jmthibault79) [inline ui/src/app/pages/runtimes-list.tsx:141]: While `slug(name)` should always produce an equal string, the actual field used for routing is workspace `id` - let's continue to use that so we don't risk these getting out of sync.

Aside: yes, `id` is a terrible name for this value and we should fix it.  (My suggestion would be `terraName`.)  That's a *big* change so let's do it in a future refactoring.

**Review by jmthibault79 — COMMENTED** [review]
I have not tested this locally yet.

Also the API tests need an update.

File: ui/src/app/pages/runtimes-list.tsx (hunk 1 above)
Comment (dmohs) [reply ui/src/app/pages/runtimes-list.tsx:141]: I checked the code for the Environments component, and the only thing we are passed is namespace and name, so we have to do our best with those values. If I were to continue working on this, I'd make a change to that component in Terra UI to pass more data so the implementor of `getLink` can make the decision about what is needed to send the user to the right location.

Comment (jmthibault79) [reply ui/src/app/pages/runtimes-list.tsx:141]: hm, ok.  Is there an easy way for me to understand what the correct parameters for `getLink()` should be?

Comment (jmthibault79) [reply ui/src/app/pages/runtimes-list.tsx:141]: optional suggestion: change the link to /analysis instead of /data

**Review by jmthibault79 — APPROVED** [review]
Tested locally - this is a very nice improvement.  Thank you for doing this.

**Comment by dmohs** [comment]
> Any ideas what, if anything, happened to cause this?

No, but I also see a null workspaceId and everything seems to work fine, so I assume it is benign.

File: ui/src/app/pages/runtimes-list.tsx (hunk 1 above)
Comment (dmohs) [reply ui/src/app/pages/runtimes-list.tsx]: Good point about `getLink()` parameters. I added [this comment](https://github.com/all-of-us/workbench/pull/8057/commits/a7ae39223ebdbddc188387faa5067593a2184b1c).

**Pull request merged**